lib/ui/painting/image_generator.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_PAINTING_IMAGE_GENERATOR_H_
+#define FLUTTER_LIB_UI_PAINTING_IMAGE_GENERATOR_H_
+
+#include "flutter/fml/macros.h"
+#include "flutter/lib/ui/painting/codec.h"
+#include "third_party/skia/include/core/SkImageInfo.h"
+
+namespace flutter {
+
+/// The minimal interface necessary for creating decoders that can be used for
+/// single-frame image decoding with optional support for decoding into a
+/// subscaled buffer.
+///
+/// See `getScaledDimensions` for details about how subpixel image decoding can
+/// be supported.
+class ImageGenerator {
```

Comment:
This class probably wants to have something like a frame count or a current frame - as in, it should support the basic operations that `BuiltinSkiaCodecGenerator` supports, right?

For now, everything but that one would always return the 0-index frame.

Reply:
At first I was trying to avoid this, but after thinking about it a bit, this is going to make things a lot simpler (especially `ImageDescriptor`).
I went ahead and added everything necessary to support multi-frame images in the protocol, which just came down to: `GetFrameCount`, `GetFrameInfo`, and two optional frame parameters added to `GetPixels`.